src/hepattn/experiments/trackml/run_filtering.py

```diff
@@ -17,8 +17,12 @@ def __init__(
         super().__init__(name, model, lrs_config, optimizer)
 
     def log_custom_metrics(self, preds, targets, stage):
-        pred = preds["final"]["hit_filter"]["hit_on_valid_particle"]
-        true = targets["hit_on_valid_particle"]
+        task = self.model.tasks[0]
+        target_field = getattr(task, "target_field", "on_valid_particle")
+        input_object = getattr(task, "input_object", "hit")
```

Comment:
I'm not sure we need `getattr`, the task should always have these attributes.

Might be worth adding `assert len(self.model.tasks) == 1` before `task = self.model.tasks[0]` though.